### PR TITLE
Labelchanged

### PR DESCRIPTION
**Pull request by rainwoodman**
I need this fix to compile my paper. Please consider merging into 2016 branch. Thanks.

### CODE REVIEW

**Comment by rainwoodman** [comment]
@stefanv -- This is more pressing than the package name stuff..

File: publisher/build_paper.py [hunk 1]
```diff
@@ -144,25 +155,16 @@ def tex2pdf(out_path):
...
                 )
-        out, err = run.communicate()
+        out1, err = run.communicate()
 
         if err:
             print("Error compiling BiBTeX")
-            return out
+            return out1, False
 
+    if "Label(s) may have changed." in out:
```
Comment (stefanv) [inline publisher/build_paper.py]: Why use out and out1 separately here?  Should we not check the second run to see if labels changed?

Comment (rainwoodman) [inline publisher/build_paper.py]: If bibtex has died, False would terminate the run, and out1, the error message is returned.

If bibtex has succeeded, we do not care its output. Whether to rerun is encoded in the output of pdflatex.

Comment (rainwoodman) [inline publisher/build_paper.py]: The second / third / ... runs will be issued by another call to tex2pdf_singlepass -- which will check the output again.

**Comment by stefanv** [comment]
I've read this patch again, and I don't think the logic is correct.  When a label change is detected, the build should return a failure, upon which you want to retry.

**Comment by rainwoodman** [comment]
I don't think the logic is incorrect; but the variable name was poorly chosen. 

The boolean indicates whether to retry. IF label has changed we need to retry. Otherwise we can break the loop. 

I have renamed the variable name from 'success' to 'retry'. Does this make more sense to you now?

**Comment by stefanv** [comment]
Thanks, Yu, I merged a slightly modified version.

**Comment by rainwoodman** [comment]
Thanks! As a reminder, could you merge this to 2016 branch if haven't done so?

**Comment by stefanv** [comment]
I did both at the same time.

**Pull request closed without merging**